org.hl7.fhir.r5/src/test/java/org/hl7/fhir/r5/test/rendering/ResourceElementTests.java

```diff
@@ -0,0 +1,226 @@
+
+    ResourceElement value = id.child("value");
+    Assertions.assertEquals("string", value.fhirType());
+    Assertions.assertEquals("value", value.name());
+    Assertions.assertEquals("Bundle.identifier.value", value.path());
+    Assertions.assertTrue(value.isPrimitive());
+    Assertions.assertTrue(value.hasPrimitiveValue());
+    Assertions.assertEquals("something2", value.primitiveValue());
+    Assertions.assertFalse(value.hasChildren());
+    Assertions.assertEquals(ElementKind.PrimitiveType, value.kind());
+
+    int i = 0;
+    for (ResourceElement link : bnd.children("link")) {
+      checkLink(i, link);
+      i++;
+    }
+
+    ResourceElement entry = bnd.child("entry");
+    Assertions.assertEquals("Bundle.entry", entry.fhirType());
+    Assertions.assertEquals("entry", entry.name());
+    Assertions.assertEquals("Bundle.entry[0]", entry.path());
+    Assertions.assertFalse(entry.isPrimitive());
+    Assertions.assertFalse(entry.hasPrimitiveValue());
+    Assertions.assertTrue(entry.hasChildren());
+    Assertions.assertEquals(ElementKind.BackboneElement, entry.kind());
+
+    ResourceElement fu = entry.child("fullUrl");
+    Assertions.assertEquals("uri", fu.fhirType());
+    Assertions.assertEquals("fullUrl", fu.name());
+    Assertions.assertEquals("Bundle.entry[0].fullUrl", fu.path());
+    Assertions.assertTrue(fu.isPrimitive());
+    Assertions.assertTrue(fu.hasPrimitiveValue());
+    Assertions.assertEquals("http://something5", fu.primitiveValue());
+    Assertions.assertFalse(fu.hasChildren());
+    Assertions.assertEquals(ElementKind.PrimitiveType, fu.kind());
+
+    ResourceElement obs = entry.child("resource");
+    checkObservation(obs);
+  }
+
+  private void checkObservation(ResourceElement obs) {
+    Assertions.assertTrue(obs.fhirType().equals("Observation"));
+    Assertions.assertEquals("resource", obs.name());
+    Assertions.assertEquals("Bundle.entry[0].resource", obs.path());
+    Assertions.assertEquals(ElementKind.BundleEntry, obs.kind());
+    
+    List<ResourceElement> children = obs.children();
+    assertEquals(3, children.size());
+
+    checkObsCode(children.get(1));
+
+    assertEquals(children.get(2), obs.child("value"));
+    assertEquals(children.get(2), obs.child("value[x]"));
+    checkObsValue(children.get(2));
+    
+    assertEquals(children.get(0), obs.child("contained"));
+    checkContained(children.get(0));
+  }
+
+  private void checkContained(ResourceElement cont) {
+    Assertions.assertEquals("Provenance", cont.fhirType());
+    Assertions.assertEquals("contained", cont.name());
+    Assertions.assertEquals("Bundle.entry[0].resource.contained[0]", cont.path());
+    Assertions.assertFalse(cont.isPrimitive());
+    Assertions.assertFalse(cont.hasPrimitiveValue());
+    Assertions.assertTrue(cont.hasChildren());
+    Assertions.assertEquals(ElementKind.ContainedResource, cont.kind());
+  }
+
+  private void checkObsValue(ResourceElement obsValue) {
```

Comment:
Should we check the value here?

Reply:
which value?

Reply:
After the merge, but I did not want to leave this hanging. Should we check the id value?

```xml
<id value="obs1"/>
```